cfg/vet.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cfg
+
+func VetConfig(c *Config) {
+	// The EnableEmptyManagedFolders flag must be set to true to enforce folder prefixes for Hierarchical buckets.
+	if c.EnableHns {
+		c.List.EnableEmptyManagedFolders = true
```

Comment:
How about the following instead of the complete if-block ?

```suggestion
		c.List.EnableEmptyManagedFolders |= c.EnableHns
```

Reply:
yeah, fair point.